hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -1894,10 +1896,12 @@ private void checkAdmin() throws OMException {
   @Override
   public void createBucket(OmBucketInfo bucketInfo) throws IOException {
     try {
-      if(isAclEnabled) {
+      if (isAclEnabled) {
         checkAcls(ResourceType.VOLUME, StoreType.OZONE, ACLType.CREATE,
             bucketInfo.getVolumeName(), bucketInfo.getBucketName(), null);
       }
+      HddsClientUtils.verifyResourceName(bucketInfo.getVolumeName(),
```

Comment:
For write requests, this code is no more used.
These changes need to bbe done in OMBucketcreateRequest#preExecute and OMVolumeCreateRequest#preExecute and the same thing needs to be done in S3BucketCreateRequest

Reply:
And if illegal, verifyResourceName throws IllegalArgumentException, here we need to catch this, and return proper error code to the client here.

Reply:
Added the checks at suggested sites.